slicedimage/backends/_http.py

```diff
@@ -26,10 +28,21 @@ def __init__(self, url, checksum_sha256, seekable):
     def __enter__(self):
         if self.seekable:
             req = requests.get(self.url)
+            if self.checksum_sha256 is not None:
```

Comment:
> Why can't this block live in the ChecksummingContextManager? Is it because it slurps all the data at once instead of digesting it by block?

If the caller requests a seekable handle, then there's no guarantee that the file will be read out sequentially.  Therefore, we must checksum it _now_.

> Does requests support adding a checksum? There seems to be something relevant here: http://flask.pocoo.org/docs/1.0/patterns/requestchecksum/

This is for flask.